### PR TITLE
[EVM] add bank precompile

### PR DESCRIPTION
**Pull request by codchen**
## Describe your changes and provide context
**This PR upgrades Go version to 1.20 to use the `embed` functionality**

Add a custom precompile to allow EVM contracts interact with bank module directly.

This PR involves a change to `go.mod` that updates go-ethereum dependency to our own fork, which extends the precompile interface to allow state being passed in.

## Testing performed to validate your change
unit test with a solidity contract that calls the bank precompile

### CODE REVIEW

File: precompiles/bank/bank.go [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+
+	return &Precompile{
+		Precompile: pcommon.Precompile{ABI: newAbi},
+		bankKeeper: bankKeeper,
+		evmKeeper:  evmKeeper,
+	}, nil
+}
+
+func (Precompile) Address() common.Address {
+	return common.HexToAddress("0x0000000000000000000000000000000000001001")
```
Comment (jewei1997) [inline precompiles/bank/bank.go]: might be good to make this part of the precompile struct.

File: precompiles/bank/bank.go [hunk 2]
```diff
@@ -0,0 +1,152 @@
+package bank
```
Comment (jewei1997) [inline precompiles/bank/bank.go:1]: should we have tests for this file?

File: precompiles/bank/Bank.sol [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+IBank constant BANK_CONTRACT = IBank(
+    BANK_PRECOMPILE_ADDRESS
+);
+
+interface IBank {
+    // Transactions
+    function send(
+        address fromAddress,
+        address toAddress,
+        string memory denom,
```
Comment (jewei1997) [inline precompiles/bank/Bank.sol:15]: for my education: what are some examples of what denom could be?

File: x/evm/keeper/testdata/SendAll.sol [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+import "../../../../precompiles/bank/Bank.sol";
+
+contract SendAll {
+    function sendAll(
+        address fromAddress,
+        address toAddress,
+        string memory denom
+    ) public {
+        uint256 amount = BANK_CONTRACT.balance(fromAddress, denom);
+        BANK_CONTRACT.send(fromAddress, toAddress, denom, amount);
```
Comment (jewei1997) [inline x/evm/keeper/testdata/SendAll.sol:13]: Can possibly emit an event here to make indexing easier in the future.

**Review by jewei1997 — APPROVED** [review]

File: precompiles/bank/Bank.sol (hunk 3 above)
Comment (codchen) [reply precompiles/bank/Bank.sol:15]: It'd be one of `usei`, `ibc/<route1>/<route2>/....` tokens, `tokenfactory/xyz...` tokens

File: precompiles/bank/bank.go (hunk 1 above)
Comment (codchen) [reply precompiles/bank/bank.go]: updated

File: precompiles/bank/bank.go (hunk 2 above)
Comment (codchen) [reply precompiles/bank/bank.go:1]: added

**Pull request merged**